guard/src/commands/validate.rs

```diff
@@ -128,64 +148,106 @@ Note - When pointing the command to a directory, the directory may not contain a
-                          \nFor directory arguments such as `param-dir1` above, scanning is only supported for files with following extensions: .yaml, .yml, .json, .jsn, .template")
-                .multiple(true))
-            .arg(Arg::with_name(TYPE.0).long(TYPE.0).short(TYPE.1).takes_value(true).possible_values(["CFNTemplate"])
-                .help("Specify the type of data file used for improved messaging"))
-            .arg(Arg::with_name(OUTPUT_FORMAT.0).long(OUTPUT_FORMAT.0).short(OUTPUT_FORMAT.1).takes_value(true)
-                .possible_values(["json", "yaml", "single-line-summary"])
+                          \nFor directory arguments such as `param-dir1` above, scanning is only supported for files with following extensions: .yaml, .yml, .json, .jsn, .template"))
+            .arg(Arg::new(TYPE.0)
+                .long(TYPE.0)
+                .short(TYPE.1)
+                .action(ArgAction::Set)
+                .required(false)
+                .help("Specify the type of data file used for improved messaging - ex: CFNTemplate"))
+            .arg(Arg::new(OUTPUT_FORMAT.0).long(OUTPUT_FORMAT.0).short(OUTPUT_FORMAT.1)
+                .value_parser(OUTPUT_FORMAT_VALUE_TYPE)
                 .default_value("single-line-summary")
+                .action(ArgAction::Set)
                 .help("Specify the format in which the output should be displayed"))
-            .arg(Arg::with_name(PREVIOUS_ENGINE.0).long(PREVIOUS_ENGINE.0).short(PREVIOUS_ENGINE.1).takes_value(false)
+            .arg(Arg::new(PREVIOUS_ENGINE.0).long(PREVIOUS_ENGINE.0).short(PREVIOUS_ENGINE.1)
+                .action(ArgAction::SetTrue)
                 .help("Uses the old engine for evaluation. This parameter will allow customers to evaluate old changes before migrating"))
-            .arg(Arg::with_name(SHOW_SUMMARY.0).long(SHOW_SUMMARY.0).short(SHOW_SUMMARY.1).takes_value(true).use_delimiter(true).multiple(true)
-                .possible_values(["none", "all", "pass", "fail", "skip"])
+            .arg(Arg::new(SHOW_SUMMARY.0)
+                .long(SHOW_SUMMARY.0)
+                .short(SHOW_SUMMARY.1)
+                .use_value_delimiter(true)
+                .action(ArgAction::Append)
+                .value_parser(SHOW_SUMMARY_VALUE_TYPE)
                 .default_value("fail")
                 .help("Controls if the summary table needs to be displayed. --show-summary fail (default) or --show-summary pass,fail (only show rules that did pass/fail) or --show-summary none (to turn it off) or --show-summary all (to show all the rules that pass, fail or skip)"))
-            .arg(Arg::with_name(SHOW_CLAUSE_FAILURES.0).long(SHOW_CLAUSE_FAILURES.0).short(SHOW_CLAUSE_FAILURES.1).takes_value(false).required(false)
+            .arg(Arg::new(SHOW_CLAUSE_FAILURES.0)
+                .long(SHOW_CLAUSE_FAILURES.0)
+                .short(SHOW_CLAUSE_FAILURES.1)
+                .action(ArgAction::SetTrue)
```

Comment:
We are missing `required()` here again with few of the args, from the documentation I couldn't tell if absence of this implies that this is a mandatory argument or not. So just pointing it out, in case it does.

Reply:
see previous comment about flags